src/lib.rs

```diff
@@ -49,557 +124,1353 @@ pub fn parse_document(buffer: &str) -> Node {
         let buffer_len = buffer.len() as size_t;
         let p_buffer = buffer.as_ptr();
         let root_ptr = cmark_parse_document(p_buffer, buffer_len, 0);
-        let manager = ResourceManager::make_shared();
-        let resource = manager.borrow_mut().resource_for(root_ptr);
-        let capabilities = CapabilityFactory::new()
-            .with_all()
-            .build(&resource, manager.clone());
+        let manager = Rc::new(ResourceManager::new());
```

Comment:
What part of this code is unsafe? Does it have to be that way? Can we narrow the scope of the unsafe block?